src/base/registrations/registrar.ts

```diff
@@ -58,7 +58,7 @@ state.subscribe((s: Connection) => {
   console.log("register.state", s);
 });
 
-export async function getRegistration(name: string, config: Config, resolver?: EnsResolver): Promise<Registration | null> {
+export async function getRegistration(name: string, config: Config, resolver?: EnsResolver | null): Promise<Registration | null> {
```

Comment:
We want to allow both `null` and `undefined`?

Reply:
The issue here is with ethers.js, on line 65 `config.provider.getResolver(name)` returns `EnsResolver | null` to `undefined` and Typescript doesn't like that.

IMO if we want to go back to only accept `EnsResolver | undefined` we could check on line 65 if `getResolver` returns null and return early in that case.

Something like

```ts
  if (! resolver) {
    const newResolver = await config.provider.getResolver(name);
    if (newResolver !== null) resolver = newResolver;

    if (! resolver) {
      return null;
    }
```

Reply:
Ok let's handle this in a different PR, I think we should slowly move away from `null` as parameters or inside datastructures (it's fine as a return value though).